azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -539,6 +549,50 @@ public int selectAndUpdateExecutionWithLocking(final int executorId, final boole
     }
   }
 
+  private static class FetchAgedQueuedExecutableFlows implements
+      ResultSetHandler<List<Pair<ExecutionReference, ExecutableFlow>>> {
+
+    // Select queued unassigned flows
+    private static final String FETCH_FLOWS_QUEUED_FOR_LONG_TIME =
+        "SELECT exec_id, enc_type, flow_data, status FROM execution_flows"
+            + " WHERE submit_time < ? AND status = "
+            + Status.PREPARING.getNumVal();
```

Comment:
If we're only interested in unassigned executions "executor_id is NULL" must be added to the WHERE clause. Without it executions that are already dispatched will be retuned as well.

Reply:
The intent is to gather both the unassigned and assigned ones.

Reply:
In that case could you rephrase the comment above the query to reflect this? It's confusing as the comment indicates something that the actual query is not fulfilling.